File: FizzBuzz/FizzBuzz/FizzBuzzIntegration.cs
namespace FizzBuzz;

public class FizzBuzzIntegration
{
    private readonly Printer _printer;

    public FizzBuzzIntegration(Printer printer)
    {
        _printer = printer;
    }

    public void FizzBuzz()
    {
        var range = RangeGenerator.GenerateRange();
        var results = range.Select(FizzBuzzConverter.Convert);
        results.ForEach(_printer.Print);
    }
}

File: clean-code/src/main/java/ch/adesso/educamp/FizzBuzzIntegration.java
package ch.adesso.educamp;

public class FizzBuzzIntegration {

    static void runFizzBuzz() {
        String[] range = RangeGenerator.generateRange();
        String[] convertedNumbers = FizzBuzzConverter.convert(range);
        Printer.print(convertedNumbers);
    }

}
